expression/expression.go

```diff
@@ -687,8 +688,106 @@ func IsBinaryLiteral(expr Expression) bool {
 	return ok && con.Value.Kind() == types.KindBinaryLiteral
 }
 
-// CheckExprPushFlash checks a expr list whether each expr can be pushed to flash storage.
-func CheckExprPushFlash(exprs []Expression) (exprPush, remain []Expression) {
+// CheckExprPushDown check if all the expr in the expr list can be push down to the storage.
+func CheckExprPushDown(sc *stmtctx.StatementContext, exprs []Expression, client kv.Client, storeType kv.StoreType) bool {
+	_, _, remained := ExpressionsToPB(sc, exprs, client)
```

Comment:
Why don't we add an input argument `storeType` for `CheckExprPushDown`? If so, we do not need `checkFlashExprPushDown` and `checkTiKVExprPushDown `

Reply:
`storeType` is already an input argument of `CheckExprPushDown`, actually the check contains two stage: first to check if the exprs can be converted to PB, which is a precondition for all store type, second to check if the exprs can be pushed based on the store type, I don't want to make `CheckExprPushDown` a recursive function because `ExpressionsToPB` check the exprs recursively, make `CheckExprPushDown` a recursive function will make many redundant calls to `ExpressionsToPB`.

Reply:
Sorry for the wrong statements before, what I meant before is that, why not add an input argument `storeType` for ExpressionToPB.

Reply:
Discussed offline, and refine the code based on the discussion.